solidity/contracts/KeepRandomBeaconOperator.sol

```diff
@@ -692,23 +692,26 @@ contract KeepRandomBeaconOperator is ReentrancyGuard {
     }
 
     /**
-     * @dev Gets all indices in the provided group for a member.
+     * @notice Return whether the given operator
+     * has withdrawn their rewards from the given group.
      */
-    function getGroupMemberIndices(bytes memory groupPubKey, address member) public view returns (uint256[] memory indices) {
-        return groups.getGroupMemberIndices(groupPubKey, member);
+    function hasWithdrawnRewards(address operator, uint256 groupIndex)
+        public view returns (bool) {
+        return groups.hasWithdrawnRewards(operator, groupIndex);
     }
 
     /**
      * @dev Withdraws accumulated group member rewards for operator
-     * using the provided group index and member indices. Once the
-     * accumulated reward is withdrawn from the selected group, member is
-     * removed from it. Rewards can be withdrawn only from stale group.
+     * using the provided group index.
+     * Once the accumulated reward is withdrawn from the selected group,
+     * the operator is flagged as withdrawn.
+     * Rewards can be withdrawn only from stale group.
      * @param operator Operator address.
      * @param groupIndex Group index.
-     * @param groupMemberIndices Array of member indices for the group member.
      */
-    function withdrawGroupMemberRewards(address operator, uint256 groupIndex, uint256[] memory groupMemberIndices) public nonReentrant {
-        uint256 accumulatedRewards = groups.withdrawFromGroup(operator, groupIndex, groupMemberIndices);
+    function withdrawGroupMemberRewards(address operator, uint256 groupIndex)
```

Comment:
The worst-case cost of iterating through the members is `128 * (800 + change)` or roughly 100k excess gas if not optimized with dynamic arrays or other better data structures. With a dynamic array optimization it would be `64 * (800 + change) ≈ 50k`. Where iterating through members becomes prohibitive is if _transfers_ are made to multiple addresses in a single call (past OOG with 64 members).

It would be possible to store this data (which members are at which indices) directly on-chain making these queries very cheap at the cost of making the cost of group member storage in DKG result submission roughly twice as high. An even more sophisticated data structure could store the information in a highly compressed `bytes` form but it'd be impossible to develop within the deadline.

If we go back to a `groupMemberIndices`-based interface we'd need to record the withdrawal status of all those members individually instead of deleting them (net 25k gas per member). We could win some codesize gains by keeping `getGroupMemberIndices` in the statistics contract, but that'd again require extra work in the dApp.